lib/core/application/event/event_datetime_settings_bloc/event_datetime_settings_bloc.dart

```diff
@@ -120,23 +193,29 @@ class EventDateTimeSettingsEvent with _$EventDateTimeSettingsEvent {
   factory EventDateTimeSettingsEvent.init({
     required DateTime startDateTime,
     required DateTime endDateTime,
+    required DateTime tempStartDateTime,
+    required DateTime tempEndDateTime,
   }) = EventDateTimeSettingsEventInit;
 
   const factory EventDateTimeSettingsEvent.startDateChanged({
     required DateTime datetime,
   }) = StartDateChanged;
 
-  const factory EventDateTimeSettingsEvent.startTimeChanged({
-    required DateTime datetime,
-  }) = StartTimeChanged;
-
   const factory EventDateTimeSettingsEvent.endDateChanged({
     required DateTime datetime,
   }) = EndDateChanged;
 
-  const factory EventDateTimeSettingsEvent.endTimeChanged({
-    required DateTime datetime,
-  }) = EndTimeChanged;
+  const factory EventDateTimeSettingsEvent.timezoneChanged({
+    required String timezone,
+  }) = TimezoneChanged;
+
+  const factory EventDateTimeSettingsEvent.saveChanges({
+    required DateTime tempStart,
```

Comment:
not sure why call saveChanges, we have to send tempStart and tempEnd again ? Aren't these two value already updated on `startDateChanged` `endDateChanged` event ?